fendermint/actors/src/manifest.rs

```diff
@@ -0,0 +1,78 @@
+// Copyright 2022-2024 Protocol Labs
+// SPDX-License-Identifier: Apache-2.0, MIT
+use anyhow::{anyhow, Context};
+use cid::Cid;
+use fvm_ipld_blockstore::Blockstore;
+use fvm_ipld_encoding::CborStore;
+use std::collections::HashMap;
+
+const CHAINMETADATA_ACTOR_NAME: &str = "chainmetadata";
+
+/// A mapping of internal actor CIDs to their respective types.
+pub struct Manifest {
+    pub chainmetadata_code: Cid,
+    by_id: HashMap<u32, Cid>,
+    by_code: HashMap<Cid, u32>,
+}
+
+pub const CHAINMETADATA_ACTOR_CODE_ID: u32 = 1;
+pub const CHAINMETADATA_ACTOR_ID: u64 = 48;
```

Comment:
Should this actor have its own module under `actor_interface` where this is declared? I'm already regretting not putting all those IDs into the `lib.rs` of that crate, so it's easier to see at a glance which IDs are free, but this for sure isn't a place I'd be looking for further IDs.

Reply:
I think we should try to move these IDs to the `fendermint_vm_actor_interface` package along with all the others, and forget the `CODE_ID` part, just make it look like one of the [ipc actors](https://github.com/consensus-shipyard/ipc/blob/8f165c6db1eec7898ad5725a541f431bc41f0437/fendermint/vm/actor_interface/src/ipc.rs#L30-L31): it should have a stable ID and address, but not a code ID. 

The code ID seems like an unstable identifier that might work for a fixed list of builtin actors, but looks like a timebomb for custom actors that might be merged with user defined ones and depend on instantiation order.

Reply:
I created a `chainmetadata` module in the `fendermint_vm_actor_interface` package where I call `define_id` macro to generate the CHAINMETADATA_ID/CHAINMETADATA_ACTOR.

Funnily, I accidentally set its id ti 64, which caused the graph-test to fail with a `failed to create checkpoint` error which took me a while to figure out as I had changed so several other things at the same time. 

So +1 for having all of these IDs in `lib.rs` so to not accidentally use the same ID twice, which when happens silently gets accepted.